### PR TITLE
Character: Private families

### PR DESCRIPTION
**Pull request by spitfire305**

### CODE REVIEW

File: app/Models/Character.php [hunk 1]
```diff
@@ -158,6 +158,15 @@ public function scopePreparedWith(Builder $query): Builder
             'families' => function ($sub) {
                 $sub->select('families.id', 'families.name');
             },
+            'characterFamilies' => function ($sub) {
+                $sub->select('*');
```
Comment (ilestis) [inline app/Models/Character.php:162]: This seems like a waste of processing power :-)

File: app/Models/Character.php [hunk 2]
```diff
@@ -234,13 +243,16 @@ public function families(): BelongsToMany
     public function characterFamilies(): HasMany
     {
         return $this->hasMany(CharacterFamily::class, 'character_id')
+            ->orderBy('id')
```
Comment (ilestis) [inline app/Models/Character.php:246]: Isn't this automatic?

File: app/Models/CharacterFamily.php [hunk 3]
```diff
@@ -4,22 +4,27 @@
...
 use Carbon\Carbon;
 use Illuminate\Database\Eloquent\Model;
+use App\Models\Concerns\Privatable;
 use Illuminate\Database\Eloquent\Relations\BelongsTo;
 
 /**
  * @property int $int
  * @property int $character_id
  * @property int $family_id
+ * @property bool $is_private
```
Comment (ilestis) [inline app/Models/CharacterFamily.php:14]: This should be added by the trait already

File: app/Models/Family.php [hunk 4]
```diff
@@ -176,7 +177,7 @@ public function datagridSelectFields(): array
     }
 
     /**
-     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
+     * @return BelongsTo
      */
     public function location()
```
Comment (ilestis) [inline app/Models/Family.php:182]: Should be a function return typing instead of a commend

File: app/Renderers/Layouts/Location/Character.php [hunk 5]
```diff
@@ -32,9 +32,16 @@ public function columns(): array
             ],
             'families' => [
                 'label' => Module::plural(config('entities.ids.family'), 'entities.families'),
-                'class' => self::ONLY_DESKTOP,
-                'render' => Standard::ENTITYLIST,
-                'with' => 'families',
+                'render' => function ($model) {
```
Comment (ilestis) [inline app/Renderers/Layouts/Location/Character.php:35]: Can't you just do `'with' => 'families'` and skip the whole custom rendering?

File: app/Renderers/Layouts/Location/Character.php [hunk 6]
```diff
@@ -47,8 +54,16 @@ public function columns(): array
             'races' => [
                 'label' => Module::plural(config('entities.ids.race'), 'entities.races'),
                 'class' => self::ONLY_DESKTOP,
-                'render' => Standard::ENTITYLIST,
-                'with' => 'races',
+                'render' => function ($model) {
```
Comment (ilestis) [inline app/Renderers/Layouts/Location/Character.php:57]: Same here, can't we just do with characterRaces ?

**Review by ilestis — CHANGES_REQUESTED** [review]
Small tweaks

File: app/Models/Character.php (hunk 2 above)
Comment (spitfire305) [reply app/Models/Character.php:246]: For some reason, it isnt.

File: app/Renderers/Layouts/Location/Character.php (hunk 5 above)
Comment (spitfire305) [reply app/Renderers/Layouts/Location/Character.php:35]: Done that way the privacy functions dont work properly, it was done like that at first for character races and it caused issues.

**Review by ilestis — APPROVED** [review]
LGTM

**Pull request merged**